File: javascript/simple-linked-list/simple-linked-list.js
export class Element {
  constructor(value) {
    this.value = value;
    this.next = null;
  }
}

export class List {
  constructor(list = []) {
    this.length = 0;
    this.head = null;
    list.forEach(value => this.add(new Element(value)));
  }

  add(element) {
    element.next = this.head;
    this.head = element;
    this.length++;
  }

  toArray() {
    const res = [];
    let curr = this.head;
    while (curr !== null) {
      res.push(curr.value);
      curr = curr.next;
    }
    return res;
  }

  reverse() {
    return new List(this.toArray());
  }
}

File: swift/simple-linked-list/Sources/SimpleLinkedList/SimpleLinkedList.swift
class Element<T> {
    var value: T?
    var next: Element?
    
    init(_ value: T? = nil, _ next: Element? = nil) {
        self.value = value
        self.next = next
    }
    
    func reverseElements() -> Element {
        var head: Element = self
        var prev: Element? = nil
        while head.next != nil {
            let next = head.next!
            head.next = prev
            prev = head
            head = next
        }
        head.next = prev
        return head
    }
    
    func toArray() -> [T] {
        var res = [T]()
        var curr = self
        
        if curr.value != nil {
            res.append(curr.value!)
        }
        
        while curr.next != nil {
            curr = curr.next!
            if curr.value != nil {
                res.append(curr.value!)
            }
        }

        return res
    }
    
    static func fromArray(_ arr: [T]) -> Element {
        var next: Element? = nil
        var new = Element(nil, nil)
        for i in arr.reversed() {
            new = Element(i, next)
            next = new
        }
        return new
    }
}
